runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -2770,9 +2770,14 @@ class MM_ObjectAccessBarrierAPI
 			if (isSATBDoubleBarrierEnabled(vmThread)) {
 				vmThread->javaVM->memoryManagerFunctions->J9WriteBarrierPre(vmThread, object, destAddress, value);
 			} else {
-				j9object_t oldObject = readObjectImpl(vmThread, destAddress, false);
-				if (NULL != oldObject) {
-					if (!isMarked(vmThread, oldObject)) {
+				j9object_t prevObject = readObjectImpl(vmThread, destAddress, false);
+				if (NULL != prevObject) {
+					/* Check to see if object is old.
+					 * if ((object - vmThread->omrVMThread->heapBaseForBarrierRange0) < vmThread->omrVMThread->heapSizeForBarrierRange0) then old
+					 */
+					UDATA objectDelta = (UDATA)prevObject - (UDATA)vmThread->omrVMThread->heapBaseForBarrierRange0;
+					UDATA size = vmThread->omrVMThread->heapSizeForBarrierRange0;
```

Comment:
since this pattern repeats a number of times, let's create isOld helper and use it throughout this file